lib/utils/discover.ts

```diff
@@ -7,34 +8,36 @@ interface LocalBalenaOsDevice {
 	port: number;
 }
 
-// Although we only check for 'balena-ssh', we know, implicitly, that balenaOS
-// devices come with 'rsync' installed that can be used over SSH.
-const avahiBalenaSshTag = 'resin-ssh';
+const avahiBalenaSshConfig = {
+	type: 'ssh',
+	name: '_resin-device._sub',
+	protocol: 'tcp' as const,
+};
+
+const avahiBalenaSshSubtype = 'resin-device';
 
 export async function discoverLocalBalenaOsDevices(
 	timeout = 4000,
 ): Promise<LocalBalenaOsDevice[]> {
-	const availableServices = await enumerateServices();
-	const serviceDefinitions = Array.from(availableServices)
-		.filter((s) => Array.from(s.tags).includes(avahiBalenaSshTag))
-		.map((s) => s.service);
-
-	if (serviceDefinitions.length === 0) {
-		throw new Error(
-			`Could not find any available '${avahiBalenaSshTag}' services`,
-		);
-	}
+	const bonjour = new Bonjour();
```

Comment:
I think it would be good to pass an error handler and report errors properly instead of letting them crash the process: https://github.com/onlxltd/bonjour-service#initializing

Reply:
I did that using the standard CLI error handling. I did force a simple error which looked fine with it.